### ISSUE TITLE
[bug]: Duplicate images are generated

### ISSUE THREAD

**Issue by GregoryCARON** [invoke-ai/InvokeAI#4688]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Windows

### GPU

cuda

### What version did you experience this issue on?

3.2.0rc2

### What happened?

When we generate an image, it is generated twice

### Screenshots

<img width="178" alt="image" src="https://github.com/invoke-ai/InvokeAI/assets/43751741/376879a8-2172-4ffa-8b0e-01ded6238e41">

**Comment by psychedelicious** [comment]
Which tab are you generating on?

If you are generating using the workflow editor, please provide the workflow so we can reproduce the issue.

**Comment by psychedelicious** [comment]
@hipsterusername There was a node that wasn't marked as intermediate in one of the graphs, was last week. That's since been fixed, and I cannot reproduce this on main. 

It's possible the latest RC doesn't have the fix; I cannot remember the specific graph that was affected or find the commit that fixed it.

Need more information to troubleshoot - settings and tab. A screenshot of all the settings panel is fine.

**Comment by GregoryCARON** [comment]
I currently do not have access to the machine where InvokeAI is installed, I will try to install that as quickly as possible

**Comment by oppenheimer-** [comment]
I have the same problem and happy to provide further info.

- I am using 3.2.0rc2 on windows with cuda as well and created a board in my gallery.
- In contrary to @GregoryCARON the duplicates don't show up in the gallery, but in a specified output directory. 
- Upon install i opted for a different output directory than the default. It's on on the same drive though.
- the duplicates have unique filenames that look like hashes
- thumbnails get duplicated as webp as well

**Comment by psychedelicious** [comment]
@oppenheimer- If you see a single image in the gallery, it's not the same problem.

The extra images are intermediate images, that is, outputs of the graph as it worked towards the final image. If you want to clear out the extra images, use the Clear Intermediates button in settings.

This behaviour is related to a recent change to allow images to be saved to a board in the workflow editor and I understand it's unexpected, but it's not a malfunction.

**Comment by oppenheimer-** [comment]
alright, thanks for the clarification and sorry for party crashing ;)

**Comment by Oflrnos** [comment]
Updated to the release v3.2.0 earlier this morning. It began creating duplicate images that persist and are the same in all aspects except ID / name upon clicking on **Info </>**. generated via Text-to-Image.

Windows. 6GB VRAM. Cuda.

And it stays after refreshing, restarting and having cleared intermediates.
in earlier version, the dupes did appear but vanished upon refreshing or having done other steps.

![image](https://github.com/invoke-ai/InvokeAI/assets/95576291/0824c108-b5eb-4178-8ecb-3fb98f568a3d)
![image](https://github.com/invoke-ai/InvokeAI/assets/95576291/d2fae11a-6203-4eb9-9b42-fdd29cb690ca)

**Comment by DavidCWGA** [comment]
A similar thing has started to happen to me after upgrading to 3.2.0. Generated images appear only once in the web UI, but there are two identical images in the "images" output directory, only with different UUIDs as filenames. The files are literally identical, they have the same hash.

"invokeai-db-maintenance" does _not_ recognize them as orphan image files, and doesn't remove them.

**Comment by hipsterusername** [comment]
It is expected in outputs - These are not orphan images, but 'intermediates' that can be cleared from settings.

It is **not** expected for them to display in the gallery twice.

**Comment by DavidCWGA** [comment]
Well you're right, clearing intermediates did remove them, but I still don't understand what an "intermediate" is when it is literally an identical copy of the final image.

**Comment by psychedelicious** [comment]
@Oflrnos 
Thanks. Can you please open a private / incognito browser window and confirm that you get duplicate images there?

**Comment by Oflrnos** [comment]
@psychedelicious it was already done within incognito. didn't want to open the dozen other windows with Invoke running. I've since reverted to 3.1.1 as I was / am doing higher quantities overnight and didn't want to sift through doubles all the time.

**Comment by DarthRuebezahl** [comment]
I have the same issue in the vanilla 3.2.0 release version. Duplicate images are generated for every single image. They are absolutely identical (I checked in a hex editor). They are removed by "Clear intermediates". This also happens in an incognito window.

3.1.0 could do without having an identical intermediate image. If this behaviour can't be reverted back, then perhaps it is time for a "Remove intermediates automatically" setting that removes intermediates when they are not needed anymore (though I guess that would not be trivial, as intermediates appear in different contexts). I have never had the need to go back to an intermediate image so far.

**Comment by DarthRuebezahl** [comment]
I have to backpedal on my previous statement: Sometimes, "clear intermediates" does NOT clear the duplicate images, so sometimes I have to wade through a few hundred images and delete the duplicates manually.

**Comment by 64jcl** [comment]
Where is this "Clear intermediates" function? I have the same issue with duplicates and it is a bit annoying.

**Comment by ufuksarp** [comment]
3.3.0post3 reporting in, still present. Same image twice in the output folder.

**Comment by hipsterusername** [comment]
> 3.3.0post3 reporting in, still present. **Exact** same image twice in the output folder.

Expected in the output folder. Not expected in the gallery. 

Can you confirm which?

**Comment by ufuksarp** [comment]
Output folder. I don't really understand why it's "expected", but yes, output folder.

**Comment by hipsterusername** [comment]
It is expected as it's part of the graph - In order to ensure that all generations go to the correct board selected at generation time, a "Save Image to Board" node is added at the end of the graph.

Ultimately, this means that the engine is storing an intermediate image for the "Latents to Image" node, and then the final asset as part of the "Save Image" node.

Clearing intermediates will remove the intermediate image stored as part of the graph. We're open to thoughts (and contributions) on how/when intermediates ought to be cleared on some automated frequency, but this can currently be done manually in the settings tab.

**Comment by hipsterusername** [comment]
Everything is nodes.
